samples/drivers/soc_flash_nrf/src/main.c

```diff
@@ -10,8 +10,18 @@
 #include <device.h>
 #include <stdio.h>
 
-/* Offset between pages */
-#define FLASH_TEST_OFFSET 0x40000
+#if defined(CONFIG_SOC_SERIES_NRF52X)
+#define FLASH_TEST_OFFSET DT_FLASH_AREA_IMAGE_1_OFFSET
+#elif defined(CONFIG_SOC_NRF9160_SICA)
+#ifdef TRUSTED_EXECUTION_NONSECURE
+#define FLASH_TEST_OFFSET DT_FLASH_AREA_IMAGE_1_OFFSET
+#else
+#define FLASH_TEST_OFFSET DT_FLASH_AREA_IMAGE_1_NONSECURE_OFFSET
+#endif
+#else
+#error target is not supported by the sample
+#endif
```

Comment:
This bunch of `#if`s is not that easy to understand. I was wondering if it could be written in a bit more readable way.
And is nRF51 intentionally not supported?

Reply:
nRF51 Was not supported before - I can add support for it within another patch (to upsterem preferably)